pandas/core/generic.py

```diff
@@ -3233,21 +3228,41 @@ def fillna(self, value=None, method=None, axis=None, inplace=False,
 
             # 3d
             elif self.ndim == 3:
-
                 # fill in 2d chunks
-                result = dict([(col, s.fillna(method=method, value=value))
-                               for col, s in self.iteritems()])
+                if axis == 0:
+                    frame = self.swapaxes(0, 1)
```

Comment:
Question (not too familiar with Panels): swapping the axes here can potentially cause dtypes to change? (ints upcast to float, or numeric upcast to object when you have mixed dtypes)